src/main.cpp

```diff
@@ -22,13 +23,18 @@ int main(int argc, char *argv[])
 
     // Command line parsing
     QCommandLineParser parser;
-    parser.setApplicationDescription("A secure translation service that performs translations for you locally, on your own machine.");
-    parser.addHelpOption();
-    parser.addVersionOption();
-    parser.process(translateLocally);
+    translateLocally::CLIArgumentInit(translateLocally, parser);
 
-    // Launch application
-    MainWindow w;
-    w.show();
-    return translateLocally.exec();
+    // Launch application unless we're supposed to be in CLI mode
+    if (translateLocally::isCLIOnly(parser)) {
+        // The CLI is supposed to just do something and then die.
+        // To do that we will just have an object whose constructor as a side
+        // effect will do the thing and die.
```

Comment:
Is this comment still true? Looks like the meaty part of the CLI is implemented implemented in `run()`, not the constructor. There don't seem to be all that many side effects to the constructor :D

Reply:
Yeah that's from earlier version when my design was more appalling.